pylint/plugins/hass_imports.py

```diff
@@ -548,59 +548,122 @@ def _visit_importfrom_relative(
         if len(split_package) < node.level + 2:
             self.add_message("hass-absolute-import", node=node)
 
+    def _check_for_constant_alias(
+        self,
+        node: nodes.ImportFrom,
+        current_component: str | None,
+        imported_component: str,
+    ) -> bool:
+        """Check for hass-import-constant-alias."""
+        if current_component == imported_component:
+            return True
+
+        # Check for `from homeassistant.components.other import DOMAIN`
+        for name, alias in node.names:
+            if name == "DOMAIN" and (alias is None or alias == "DOMAIN"):
+                self.add_message(
+                    "hass-import-constant-alias",
+                    node=node,
+                    args=(
+                        "DOMAIN",
+                        "DOMAIN",
+                        f"{imported_component.upper()}_DOMAIN",
+                    ),
+                )
+                return False
+
+        return True
+
+    def _check_for_component_root_import(
+        self,
+        node: nodes.ImportFrom,
+        current_component: str | None,
+        imported_parts: list[str],
+        imported_component: str,
+    ) -> bool:
+        """Check for hass-component-root-import."""
+        if (
+            current_component == imported_component
+            or imported_component in _IGNORE_ROOT_IMPORT
+        ):
+            return True
+
+        # Check for `from homeassistant.components.other.module import something`
+        if len(imported_parts) > 3:
+            self.add_message("hass-component-root-import", node=node)
+            return False
+
+        # Check for `from homeassistant.components.other import const`
+        for name, _ in node.names:
+            if name == "const":
+                self.add_message("hass-component-root-import", node=node)
+                return False
+
+        return True
+
+    def _check_for_relative_import(
```

Comment:
This sounds like it could be a ruff rule, as it doesn't seem specific to Home Assistant.

Reply:
As discussed, I'll take a look at `TID252` rule in a follow-up to see if/what it could replace.
https://docs.astral.sh/ruff/rules/relative-imports/